java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -2743,6 +2745,19 @@ public static Table fromPackedTable(ByteBuffer metadata, DeviceMemoryBuffer data
     return result;
   }
 
+  /**
+   * Gather `n` samples from table randomly
+   * The output is not same with CPU Sample exec, but this is faster.
+   *
+   * @param n non-negative number of samples expected
+   * @param replacement Allow or disallow sampling of the same row more than once.
+   * @param seed Seed value to initiate random number generator.
+   * @return
+   */
+  public Table sample(long n, boolean replacement, long seed) {
```

Comment:
Shall we need to add tests to check at least whehter the `n` works ? 
And what will happen if `n` < 0 ?

Reply:
Cudf code:   CUDF_EXPECTS(n >= 0, "expected number of samples should be non-negative");
Our code will not generate negative number.
val numSampleRows = (cb.numRows() * (upperBound - lowerBound)).toLong
upperBound > lowerBound

Reply:
How about adding tests to check at least whehter the `n` works ?

Reply:
How to know whether the `replacement` takes affect ?  What's difference between `replacement = false` and `replacement = true` ?

Reply:
Added test case.